### PR TITLE
feat(performance): Adds queues instrumentation for more platforms

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds custom queues instrumentation docs for java, go, ruby, and dotnet

### CODE REVIEW

**Comment by sl0thentr0py** [comment]
hey @edwardgou-sentry sorry for the double work but I just added all the ruby modules here, so can you remove them from this PR?
https://github.com/getsentry/sentry-docs/pull/10245

**Comment by edwardgou-sentry** [comment]
> hey @edwardgou-sentry sorry for the double work but I just added all the ruby modules [here](https://github.com/getsentry/sentry-docs/pull/10245), so can you remove them from this PR?

removed, thanks @sl0thentr0py !

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    span.setData("messaging.message.body.size", message.getBody().getBytes(StandardCharsets.UTF_8).length);
+    span.setData("messaging.message.receive.latency", latency.toMillis());
+    span.setData("messaging.message.retry.count", 0);
+
+    try {
+        // Process the message
+        processMessage(message);
+    } catch (Exception e) {
+        // In case of an error set the status to "internal_error"
+        span.setStatus(SpanStatus.INTERNAL_ERROR);
```
Comment (adinauer) [inline docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:99]: Should we rethrow here?

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+}
+
+transaction.finish();
```
Comment (adinauer) [inline docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:103]: This should go into a `finally {}` block.

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+String message = "Hello World!";
+String messageId = "abc123";
+
+// Create transaction
+ITransaction transaction = Sentry.startTransaction("queue_producer_transaction", "function");
+try {
+    final SentryTraceHeader traceparent = Sentry.getTraceparent();
+    final BaggageHeader baggage = Sentry.getBaggage();
+    
+    ISpan span = transaction.startChild("queue.publish", "queue_producer")
```
Comment (adinauer) [inline docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:37]: we should finish the span in a `finally {}` block

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    span.setData("messaging.message.id", messageId);
+    span.setData("messaging.destination.name", queue);
+    span.setData("messaging.message.body.size", message.getBytes(StandardCharsets.UTF_8).length);
+
+    // Publish the message to the queue (including current time stamp)
+    long now = Instant.now().getEpochSecond();
+    
+    connection.publish(queue, message, now, traceparent, baggage);
+}
+transaction.finish();
```
Comment (adinauer) [inline docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:48]: we should finish the transaction in a `finally {}` block

**Review by adinauer — APPROVED** [review]
Thanks for taking care of this! Left some comments for the Java parts. Improving API around continuing a trace is on my list.

I've created a follow up PR to address some of my comments: https://github.com/getsentry/sentry-docs/pull/10260, feel free to merge it in if you think it makes sense.

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+Instant now = Instant.now();
+Instant messageTime = Instant.ofEpochSecond(message.getTimestamp());
+Duration latency = Duration.between(messageTime, now);
+
+// Create transaction
+final TransactionContext transactionContext = Sentry.continueTrace(sentryTraceHeader, baggageHeader);
+ITransaction transaction = Sentry.startTransaction(transactionContext, "queue_consumer_transaction", "function");
+
+try {
+    ISpan span = transaction.startChild("queue.process", "queue_consumer")
```
Comment (adinauer) [inline docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:86]: This span is never finished, we should finish it in a `finally {}` block.

File: docs/platforms/dotnet/common/performance/instrumentation/custom-instrumentation/queues-module.mdx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+|:--|:--|:--|
+| `messaging.message.id ` | string | The message identifier |
+| `messaging.destination.name` | string | The queue or topic name |
+| `messaging.message.body.size` | number | Size of the message body in bytes |
+| `messaging.message.retry.count ` | number | The number of times a message was attempted to be processed |
+| `messaging.message.receive.latency ` | number | The time in milliseconds that a message awaited processing in queue |
+
+Your `queue.process` span must exist inside a transaction in order to be recognized as a consumer span. If you are using a <PlatformLink to="/integrations/#web-frameworks">supported web framework</PlatformLink>, the transaction is created by the integration. If you use plain Python, you can start a new one using `SentrySdk.StartTransaction()`.
+
+Use `SentrySdk.ContinueTrace()` to connect your consumer spans to their associated producer spans, and `span.SetExtra()` to mark the trace of your message as success or failed.
```
Comment (adinauer) [inline docs/platforms/dotnet/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:79]: When playing around we noticed that e.g. for JMS it's not possible to set `sentry-trace` as it doesn't allow the `-` character. Should we add a note like:

"Note: If you are using JMS, please transport the `sentry-trace` header with a different name like `sentry_trace` as the dash/hyphen/`-` isn't allowed in JMS."

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx (hunk 5 above)
Comment (adinauer) [reply docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:86]: addressed in https://github.com/getsentry/sentry-docs/pull/10260

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx (hunk 1 above)
Comment (adinauer) [reply docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:99]: addressed in https://github.com/getsentry/sentry-docs/pull/10260

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx (hunk 2 above)
Comment (adinauer) [reply docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:103]: addressed in https://github.com/getsentry/sentry-docs/pull/10260

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx (hunk 4 above)
Comment (adinauer) [reply docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:48]: addressed in https://github.com/getsentry/sentry-docs/pull/10260

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx (hunk 3 above)
Comment (adinauer) [reply docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:37]: addressed in https://github.com/getsentry/sentry-docs/pull/10260

File: docs/platforms/dotnet/common/performance/instrumentation/custom-instrumentation/queues-module.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply docs/platforms/dotnet/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:79]: Added a note:
image

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx [hunk 7]
```diff
@@ -0,0 +1,104 @@
+---
+title: Instrument Queues
+sidebar_order: 3000
+description: "Learn how to manually instrument your code to use Sentry's Queues module. "
+---
+Sentry comes with automatic instrumentation for the most common messaging queue systems. In case yours isn't supported, you can still instrument custom spans and transactions around your queue producers and consumers to ensure that you have performance data about your messaging queues.
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:6]: Updated! I took out the automatic instrumentation mention

File: docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+| Data Attribute | Type | Description |
+|:--|:--|:--|
+| `messaging.message.id ` | string | The message identifier |
+| `messaging.destination.name` | string | The queue or topic name |
+| `messaging.message.body.size` | number | Size of the message body in bytes |
+| `messaging.message.retry.count ` | number | The number of times a message was attempted to be processed |
+| `messaging.message.receive.latency ` | number | The time in milliseconds that a message awaited processing in queue |
+
+Your `queue.process` span must exist inside a transaction in order to be recognized as a consumer span. If you are using a supported framework, the transaction is created by the integration. If you are using plain Java, you can start a new one using `Sentry.startTransaction()`.
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:78]: 👍 cleaned it up 
image

**Comment by edwardgou-sentry** [comment]
I'm going to merge this PR for now! We can handle any further changes to the platform docs in new PRs

**Pull request merged**